sources/web/datalab/static/datalab.js

```diff
@@ -267,6 +267,29 @@ function initializeNotebookApplication(ipy, notebook, events, dialog, utils) {
     notebook.prototype.save_notebook = function() {
       originalSave.apply(this, /* check_lastmodified */ [ false ]);
     }
+
+    // A replacement notebook copy function that makes copies in the root if
+    // the source is under datalab/.
+    notebook.prototype.copy_notebook = function() {
+      var that = this;
+      var base_url = this.base_url;
+      var w = window.open('', IPython._target);
+      var parent = utils.url_path_split(this.notebook_path)[0];
+      if (parent.startsWith('datalab/')) {
+        parent = '';
```

Comment:
This would result in /datalab/samples/BigQuery/n.ipynb going to /samples/BigQuery/n.ipynb if I understand it right.

Would completely stripping off the path, so its just n.ipynb make more sense?

At some level, I feel this introduces or rather only worsens the impact of another shortcoming - no ability to move files within the UI. So now this creates files elsewhere where I might not want them to go.